mathesar/api/serializers/shared_serializers.py

```diff
@@ -115,105 +112,19 @@ class NumberDisplayOptionSerializer(MathesarErrorMessageMixin, OverrideRootParti
-        if 'z' in display_format.lower():
-            raise serializers.ValidationError(
-                "Timestamp without timezone column cannot contain timezone display format"
-            )
-
-
-class DateFormatValidator(AbstractDateTimeFormatValidator):
-
-    def validate(self, datetime_obj, display_format, serializer_field):
-        date_obj = arrow.get('2013-09-30')
-        if datetime_obj.time() != date_obj.time():
-            raise serializers.ValidationError("Date column cannot contain time or timezone display format")
-
-
-class TimeWithTimeZoneFormatValidator(AbstractDateTimeFormatValidator):
-
-    def validate(self, datetime_obj, display_format, serializer_field):
-        time_only_format = 'HH:mm:ssZZ'
-        time_str = arrow.get('2013-09-30T15:34:00.000-07:00').format(time_only_format)
-        parsed_time_str = arrow.get(time_str, time_only_format)
-        if parsed_time_str.date() != datetime_obj.date():
-            raise serializers.ValidationError("Time column cannot contain date display format")
-
-
-class TimeWithoutTimeZoneFormatValidator(TimeWithTimeZoneFormatValidator):
-
-    def validate(self, datetime_obj, display_format, serializer_field):
-        if 'z' in display_format.lower():
-            raise serializers.ValidationError("Time without timezone column cannot contain timezone display format")
-        return super().validate(datetime_obj, display_format, serializer_field)
+def is_time_format_valid(format):
+    if isinstance(format, str) and len(format) <= 255:
+        return True
+    else:
+        raise serializers.ValidationError("Time format string not a string or longer than 255 characters.")
 
 
-class DateDisplayOptionSerializer(MathesarErrorMessageMixin, OverrideRootPartialMixin, serializers.Serializer):
-    format = serializers.CharField(validators=[DateFormatValidator()])
-
-
-class TimestampWithoutTimezoneDisplayOptionSerializer(
+class TimeFormatDisplayOptionSerializer(
     MathesarErrorMessageMixin,
     OverrideRootPartialMixin,
     serializers.Serializer
 ):
-    format = serializers.CharField(validators=[TimestampWithoutTimeZoneFormatValidator()])
-
-
-class TimestampWithTimezoneDisplayOptionSerializer(
-    MathesarErrorMessageMixin,
-    OverrideRootPartialMixin,
-    serializers.Serializer
-):
-    format = serializers.CharField(validators=[TimestampWithTimeZoneFormatValidator()])
-
-
-class TimeWithTimezoneDisplayOptionSerializer(
-    MathesarErrorMessageMixin,
-    OverrideRootPartialMixin,
-    serializers.Serializer
-):
-    format = serializers.CharField(validators=[TimeWithTimeZoneFormatValidator()])
-
-
-class TimeWithoutTimezoneDisplayOptionSerializer(
-    MathesarErrorMessageMixin,
-    OverrideRootPartialMixin,
-    serializers.Serializer
-):
-    format = serializers.CharField(validators=[TimeWithoutTimeZoneFormatValidator()])
+    format = serializers.CharField(validators=[is_time_format_valid])
```

Comment:
It would be better to use the `max_length` parameter instead of custom validator, this would help up to throw errors with proper error codes.

Reply:
Fixed https://github.com/centerofci/mathesar/pull/1093/commits/27f75caa9a17270e0dbfed119001286ec401c181